include/swift/SILOptimizer/Utils/ValueLifetime.h

```diff
@@ -79,6 +79,9 @@ class ValueLifetimeAnalysis {
   /// instructions of the frontier that are not in the critical edges. Note that
   /// the method getCriticalEdges can be used to retrieve the critical edges.
   ///
+  /// An edge is also considered as "critical" if it has a single precedessor
+  /// but the predecessor's terminal instruction is a user the value.
```

Comment:
Can you elaborate on this here in the comment. The comment is defining something without explaining why. An example would be useful.